content/TruenasApps/UsingCustomApp.md

```diff
@@ -48,138 +35,1199 @@ If you are unsure about any configuration settings, review the [Install Custom A
+
+   Enter in **Server** the IP address for the SMB server, for example *192.168.1.100*.
+   This can be the TrueNAS host.
+
+   Enter the name of the SMB share in **Path**, for example *my-share*.
+
+   Enter the **Username** and **Password** of an account with permission to access the SMB share.
+
+   Select whether to mount the share volume as **Read Only** or enter a directory services **Domain** as needed.
+   {{< /expand >}}
+
+   {{< expand "Tmpfs (Temporary directory created on the RAM)" "v" >}}
+   To mount a temporary memory-backed storage directory, select **Tmpfs (Temporary directory created on the RAM)**.
+
+   {{< trueimage src="/images/SCALE/Apps/InstallCustomAppAddMemoryBackedVol.png" alt="Tmpfs Settings" id="Tmpfs Settings" >}}
+
+   Enter in **Mount Path** the <file>**path/to/directory**</file> where the temporary directory mounts inside the container.
+
+   Enter in **Tmpfs Size Limit (in Mi)** the maximum size of the temporary directory in mebibytes or accept the default of *500*.
+   {{< /expand >}}
+
+8. {{< include file="/static/includes/apps/AppInstallWizardResourceConfig.md" >}}
+
+9. Use **GPU Configuration** to allocate GPU resources if available and required for the application.
+
+10. Click **Install** to deploy the container.
     If you correctly configured the app, the widget displays on the **Installed Applications** screen.
 
-    When complete, the container becomes active. If the container does not automatically start, click **Start** on the widget.
+    When complete, the container becomes active.
+    If the container does not automatically start, click **Start** on the widget.
 
     Click on the App card reveals details.
 
-## Setting up App Storage
+## Installing via YAML
 
-### Defining Host Path Volumes
+{{< include file="/static/includes/apps/YAMLWarning.md" >}}
 
-You can mount TrueNAs storage locations inside the container.
-To mount TrueNAS storage, define the path to the system storage and the container internal path for the system storage location to appear.
-You can also mount the storage as read-only to prevent using the container to change any stored data.
-For more details, see the [Kubernetes hostPath documentation](https://kubernetes.io/docs/concepts/storage/volumes/#hostpath).
+To install a third-party application using a Docker Compose YAML file, go to **Apps > Discover**.
+Click <i class="material-icons" aria-hidden="true" title="more_vert">more_vert</i> then select **Install via YAML** to open the **Add Custom App** screen.
+
+{{< trueimage src="/images/SCALE/Apps/InstallCustomAppYAML.png" alt="Install Custom App via YAML" id="Install Custom App via YAML" >}}
+
+The **Add Custom App** screen contains two fields.
+
+Enter a name for the application in **Name** using lowercase, alphanumeric characters.
+
+Enter the Compose YAML file in **Custom Config**.
+
+Ensure the Compose file begins with a top-level element, such as `name:` or `services:`.
```

Comment:
I hate when we say "ensure". If you are telling them what to do here, which you are, you don't need to say ensure. Thoughts?
```suggestion
Begin the Compose file with a top-level element, such as `name:` or `services:`.
```